routes/login.js

```diff
@@ -1,77 +1,147 @@
 const express = require('express');
+const Client = require('authy-client').Client;
+const CONFIG = require('../config');
+
+const authy = new Client({ key: CONFIG.API_KEY });
 const generateToken = require('../utils/generate-token');
 
 const router = express.Router();
 const User = require('../models/user');
-const connect = require('../dbconnect');
 
 
 router.route('/login').post((req, res, next) => {
-  connect
-    .then(() => {
-
-      User
-        .findOne({ name: req.body.name })
-        .populate({
-          path: 'chats',
-          populate: {
-            path: 'messages',
-            populate: 'user'
+  const { name } = req.body;
+
+  User
+    .findOne({ name })
+    .populate({
+      path: 'chats',
+      populate: {
+        path: 'messages',
+        populate: { path: 'user' }
+      }
+    })
+    .then((user) => {
+
+      user.comparePassword(req.body.password,
+        (err, isMatch) => {
+          if (err) {
+            throw err;
           }
-        })
-        .then(user => {
-          user.comparePassword(req.body.password, function (err, isMatch) {
-            if (err) {
-              throw err;
-            }
 
-            if (isMatch) {
-              const { token, refreshToken } = generateToken(user);
+          if (isMatch) {
 
-              res.json({
-                ...JSON.parse(JSON.stringify(user)),
-                _token: token,
-                refreshToken,
-                success: true,
-                message: 'Authentication successful!',
-              });
+            res.json({
+              ...JSON.parse(JSON.stringify(user)),
+              success: true
+            });
 
-            } else {
-              return next(new Error('err'));
-            }
-          });
-        })
-        .catch((error) => {
-          return next(error);
+          } else {
+            return next(new Error('err'));
+          }
         });
-    });
+    })
+    .catch(e => console.log(e));
+
 });
 
-router.route('/registration').post((req, res, next) => {
-  const newUser = new User({
-    name: req.body.name,
-    password: req.body.password
+router.route('/registration').post((req, res) => {
+  const { name, password, email } = req.body;
+
+  User.findOne({ name }).exec((err, user) => {
+    if (err) {
+      res.json(err);
+      return;
+    }
+
+    if (user) {
+      res.json({ err: "Username Already Registered" });
+      return;
+    }
+
+    authy.registerUser({
+      countryCode: req.body.countryCode,
+      email,
+      phone: req.body.phoneNumber
+    }, (err, regRes) => {
+      if (err) {
+        res.json(err);
+        return;
+      }
+
+      const newUser = new User({
+        name,
+        password,
+        email,
+        authyId: regRes.user.id
+      });
+
+      newUser.save((err, usr) => {
```

Comment:
bad naming, we can use `response` word for that, but it also too bad, but not `usr`